static/js/settings_org.js

```diff
@@ -103,6 +103,42 @@ exports.populate_auth_methods = function (auth_methods) {
     }
 };
 
+var realm_notif_stream_label;
+var realm_notif_stream;
+
+exports.update_notifications_stream = function (stream_id) {
+    realm_notif_stream.attr("data-selected-stream-id", stream_id);
+
+    var currrent_notif_stream = stream_data.get_sub_by_id(stream_id);
+    if (currrent_notif_stream) {
+        realm_notif_stream_label.text(i18n.t("Current notifications stream: __stream_name__",
+                                             {stream_name: currrent_notif_stream.name}));
+    } else {
+        realm_notif_stream_label.text(i18n.t("Notifications stream not selected!"));
+    }
+};
+
+exports.populate_notifications_stream_dropdown = function (stream_list) {
+    var dropdown_list_body = $(".dropdown-list-body").expectOne();
+
+    list_render(dropdown_list_body, stream_list, {
+        name: "admin-realm-dropdown-stream-list",
+        modifier: function (item) {
+            return templates.render("admin-realm-dropdown-stream-list", { stream: item });
+        },
+        filter: {
+            element: $(".dropdown-search > input[type=text]"),
+            callback: function (item, value) {
+                return item.name.toLowerCase().match(value);
+            },
+        },
+    }).init();
+
+    $(".dropdown-search").click(function (e) {
+        e.stopPropagation();
+    });
+};
```

Comment:
Any input in the search dropdown should be being cleared if you exit and return later, but isn't.